src/device/runtime.jl

```diff
@@ -104,8 +104,33 @@ function report_exception_frame(idx, func, file, line)
     return
 end
 
+if VERSION >= v"1.2.0-DEV.512"
```

Comment:
Might want to think about if HSA has a better mechanism for this.

Reply:
I don't even know if they have a defined exception mechanism. I can investigate this.